### PR TITLE
docs: add envoy gateway api section

### PR DESCRIPTION
**Pull request by samos667**
This a sample of how I've implemented it for now. This not perfect, (about SecurityPolicy spec accuracy and doc too), but I'll really appreciate to refine it.

For the documentation, I don't know if I need to go further about CR declaration and explanation. Don't hesitate to tell me how to redact this.

### CODE REVIEW

**Comment by samos667** [comment]
> ### ✅ Deploy Preview for _authelia-staging_ ready!
> Name	Link
> 🔨 Latest commit	[efb43de](https://github.com/authelia/authelia/commit/efb43de34085f75541f80c7e0e6f45e13e046bfb)
> 🔍 Latest deploy log	https://app.netlify.com/sites/authelia-staging/deploys/67352c2bf6dadc0008d09653
> 😎 Deploy Preview	https://deploy-preview-8262--authelia-staging.netlify.app
> 📱 Preview on mobile	
> Toggle QR Code...
> _To edit notification comments on pull requests, go to your [Netlify site configuration](https://app.netlify.com/sites/authelia-staging/configuration/notifications#deploy-webhooks)._

 Wow what a banger flow !!

**Comment by samos667** [comment]
For the lore btw
image

File: docs/content/integration/proxies/envoy.md [hunk 1]
```diff
@@ -319,6 +319,108 @@ layered_runtime:
...
+        namespace: authelia
+        port: 80
+      path: /api/authz/ext-authz/
+  targetRefs:
+  - group: gateway.networking.k8s.io
+    kind: HTTPRoute
+    name: example
+```
+
+If the route namespace differs from the authelia service namespace, there is a need to declare a ReferenceGrant:
```
Comment (james-d-elliott) [inline docs/content/integration/proxies/envoy.md:355]:
```suggestion
If the route namespace differs from the authelia service namespace, there is a need to declare a ReferenceGrant:

```

File: docs/content/integration/proxies/envoy.md [hunk 2]
```diff
@@ -319,6 +319,108 @@ layered_runtime:
...
+        namespace: authelia
+        port: 80
+      path: /api/authz/ext-authz/
+  targetRefs:
+  - group: gateway.networking.k8s.io
+    kind: Gateway
+    name: example
+```
+
+If the gateway namespace differs from the authelia service namespace, there is a need to declare a ReferenceGrant:
```
Comment (james-d-elliott) [inline docs/content/integration/proxies/envoy.md:405]:
```suggestion
If the gateway namespace differs from the authelia service namespace, there is a need to declare a ReferenceGrant:

```

**Review by james-d-elliott — CHANGES_REQUESTED** [review]

**Review by james-d-elliott — APPROVED** [review]

**Comment by james-d-elliott** [comment]
Thanks for this! @all-contributors please add @samos667 for docs.

**Pull request merged**